server/bin/pbench-cull-unpacked-tarballs.py

```diff
@@ -327,6 +328,10 @@ def main(options):
 
     logger = get_pbench_logger(_NAME_, config)
 
+    # We're going to need the Postgres DB to track dataset state, so setup
+    # DB access.
+    Database.init_db(config, logger)
```

Comment:
Does this need to be modified per Nikhil's recent work?  (If not, does it need a big ol' comment explaining why it doesn't conform to the requirements of using the common entry point?)

Reply:
It feels weird that this line needs to be added here when no other modifications are required to this file.  Either this change was unnecessary, or something that this file already calls has now placed a new requirement on its callers.  If it's the latter, shouldn't that requirement be encapsulated inside the called module (e.g., via a module initializer or some other singleton implementation) and not imposed on existing callers?

Ditto for `server/bin/*.py`.

Reply:
Yes, this will need to be changed once Nikhil's change it in. (Or ideally when he merges this, if this one goes in first.) For now, this is hos it's done.

Beyond that, yes, I've often thought that our whole config setup is broken because of the dependency on the environment variable _or_ a `-C` option; basically, we can only get at it from the outer layer, which means it needs to be passed in.

And, yes, I have considered making a `PbenchServerConfig` singleton which can then be referenced from anywhere without passing it in ... I haven't quite gotten around to tackling that. (Not to mention schools of thought, which seem common in Python, that singletons are just hidden globals and therefore bad... a point of view that's not entirely incorrect.)

I'd like to be able to have inner code, for example, grab their own `Logger` ... but as our logging infrastructure requires a `PbenchServerConfig`, which requires this outer context, we end up doing things like unnaturally hanging loggers off classes.

You're welcome to tackle cleaning this up. I'm not doing it here!

Reply:
> basically, we can only get at it from the outer layer, which means it needs to be passed in.

Ah.  OK.  If/when this code gets Click-ified, perhaps we can put the database init in a common location (on the outside) and only have to do it once.

Re singletons, they are, indeed, very much like hidden globals...but I'm not sure that that means they are necessarily bad.  :-)